scvi/dataset/dataset.py

```diff
@@ -120,11 +120,12 @@ def update_genes(self, subset_genes):
         if hasattr(self, 'gene_symbols'):
             self.gene_symbols = self.gene_symbols[subset_genes]
         self.nb_genes = self.X.shape[1]
+        self.update_cells(np.array(self.X.sum(axis=1)>0).ravel())
 
     def update_cells(self, subset_cells):
         new_n_cells = len(subset_cells) if subset_cells.dtype is not np.dtype('bool') else subset_cells.sum()
         print("Downsampling from %i to %i cells" % (len(self), new_n_cells))
-        for attr_name in ['_X', 'local_means', 'local_vars', 'labels', 'batch_indices']:
+        for attr_name in ['_X', 'labels', 'batch_indices', 'local_means', 'local_vars']:
```

Comment:
Does the reordering matter?

Reply:
No it doesn't, it's because I had suppressed some and didn't think about the order when I rewrited them in the code